src/ItemOverlay/ItemOverlay.js

```diff
@@ -84,8 +84,10 @@ var ItemOverlay = module.exports = kind(
 	/**
 	* @private
 	*/
-	autoHideChanged: function() {
-		this.addRemoveClass('auto-hide', this.get('autoHide'));
+	autoHideChanged: function(was, is) {
+		if (this.autoHide) {
+			this.set('showing', (was || this.spotFocused));
```

Comment:
Also, if the intent is to only hide when we are enabling `autoHide` (and not vice versa as the overlay should remain hidden until focused), we could simplify this to

`if (this.autoHide) this.set('showing', false);`

as we know `was` would necessarily have a value of `false` in this case, since it is a change handler. We would still need to account for the case of the overlay currently being focused, which I am guessing was the intent of `this.spotFocused`, but that property is never set anywhere.